### ISSUE TITLE
Converting to .npy format for phy fails when spatial whitening disabled

### ISSUE THREAD

**Issue by mspacek** [spyking-circus/spyking-circus#13]
I don't want to do any whitening on my raw data, so I've set both temporal and spatial whitening to False in my .params file. After sorting in multi-file mode and then trying to convert the results to .npy for using in phy, I get this error:
```
-------------------------  Informations  -------------------------
| Exporting data for the phy GUI with 1 CPUs...
------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/local/bin/spyking-circus-subtask", line 11, in <module>
    load_entry_point('spyking-circus', 'console_scripts', 'spyking-circus-subtask')()
  File "/home/mspacek/src/spyking-circus/circus/scripts/subtask.py", line 33, in main
    circus.launch(task, filename, nb_cpu, nb_gpu, use_gpu, extension=extension)
  File "/home/mspacek/src/spyking-circus/circus/__init__.py", line 13, in launch
    module.main(filename, params, nb_cpu, nb_gpu, use_gpu, extension)
  File "/home/mspacek/src/spyking-circus/circus/converting.py", line 206, in main
    numpy.save(os.path.join(output_path, 'whitening_mat'), load_data(params, 'spatial_whitening').astype(numpy.double))
  File "/home/mspacek/src/spyking-circus/circus/shared/files.py", line 1237, in load_data
    spatial = numpy.ascontiguousarray(myfile.get('spatial')[:])
TypeError: 'NoneType' object has no attribute '__getitem__'
```
Then when I try to run phy using `circus-gui-python`:

`IOError: No such file or directory: 'whitening_mat.npy'`

I guess spyking-circus cannot currently export this matrix when whitening is disabled. However, perhaps it should export something like the identity matrix for compatibility with phy? I'll see if I can get this to work...

**Comment by mspacek** [comment]
I found that saving a simple identity matrix with
```
numpy.eye(nb_channels, dtype=numpy.double)
```
to a `whitening_mat.npy` file did the trick. But, I'm not sure where it would be best to do this in the code.

Should this identity matrix be saved to the `.basis.hdf5` file in `whitening.py` when `do_spatial_whitening` is `False` (~line 175)?

Or should this identity matrix only be generated in `converting.py` (~line 206) or perhaps in `files.py` in the `load_data()` function (~line 1234)?

Might be best to least this to you Pierre.

**Comment by yger** [comment]
I would suggest to simply export an identity matrix for phy, so only during the converting procedure. Otherwise, this would slow down a bit the code by doing useless products with an identity matrix while loading chunks. I'll make a fix asap

**Comment by yger** [comment]
In fact, this was fixed in 0.5 branch, still not yet fully stable, but I also pushed a fix in current master. This should now work

**Comment by mspacek** [comment]
Great, thanks!
